### DISCUSSION TITLE
is it possible to nest objects / collections?

### DISCUSSION THREAD

**Discussion by mneveroff** [i18next/i18next#2223] [Q&A]
The library allows us to do incredible things like:

```
"text1": "well $t(text2)",
"text2" "damn"
```
Which when we call `t(text1)` will resolve to `well damn`

But what about nesting collections? Consider the following:

```
"props": "$t(sharedProps)",
"sharedProps": {
  "name": "User Name"
}
```

When I'm debugging, `const { t } = useTranslation()`, I do:
`t('props')` and it does return `{name: 'User Name'}` but when I run `t('props.name`)` I'm getting missingKey? Whereas `t(sharedProps.name)` returns User Name? 

<img width="590" alt="image" src="https://github.com/user-attachments/assets/87fc60ee-e705-4398-a7fd-8b40622683b1">

**Comment by ahrjarrett** [comment]
@mneveroff I imagine this would work?

```javascript
t('props').name
```

What you're asking for might be theoretically possible, but implementing it would create a performance penalty for every call to `t` for all users, since it would require the translator to attempt to resolve every segment, then see if the resolved value was a virtual lookup, attempt the virtual lookup, then start the loop over.
